### ISSUE TITLE
Remove is authenticated check in with_user decorator

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#5017]
In https://github.com/mozilla/addons-server/blob/6e48cd0bf96af266e8ff349ba6b8d248a4d3c86b/src/olympia/accounts/views.py#L199-L202 we're redirecting the user with a message saying they are already logged in if they try to go through the `with_user` decorator.

In https://github.com/mozilla/addons-frontend/issues/2139 because we changed the API token cookie name, the frontend thinks the user is not connected, calls the authenticate view, but lands on that error because the user has a valid authenticated session (and `sessionid` cookie) already.

We do not want to re-send a new API token in that scenario, because that would mean you could get a fresh API token for another 30 days even though your session was about to end.

@kumar303 and I were wondering if we could just not do that check. Or rather, only do it for the register view, and not in the `with_user` decorator. If the clients hits a login view, with a valid fxa answer, then proceeding with logging them in no matter what does not seem like a bad thing ? 

cc @mstriemer

**Comment by kumar303** [comment]
Here are the steps to reproduce it in a fresh browser session which may help in illustrating how it's a problem for the frontend: https://github.com/mozilla/addons-frontend/issues/2157

**Comment by mstriemer** [comment]
So everyone is getting logged out when the cookie type changes I guess? Is this the problem?

1. User is suddenly logged out (jwt_api_auth_token is no longer used and no api_auth_token is set).
2. User logs in again.
3. AuthenticateView checks user data in request.session and sees the user is logged in.
4. Error message about already being logged in.

If that's the case it seems like the cookies being out of sync might be the actual problem.

Removing that check would fix not being able to log in, but it seems like the problem still exists. We want the two cookies to be in sync and if they aren't then we've got a problem. If I understand it correctly, running into this would mean that you look logged in on addons-server but addons-frontend doesn't see you as logged in.


When we see a user that has a `jwt_api_auth_token` then maybe it should get converted into the new format with the same expiration and set as `api_auth_token`. That does seem like it could be tricky but it would keep everyone signed in.

Another alternative would be to entirely log the user out if they don't have an `api_auth_token`.

**Comment by diox** [comment]
> If that's the case it seems like the cookies being out of sync might be the actual problem.

Yes.

> Removing that check would fix not being able to log in, but it seems like the problem still exists. We want the two cookies to be in sync and if they aren't then we've got a problem. If I understand it correctly, running into this would mean that you look logged in on addons-server but addons-frontend doesn't see you as logged in.

That's correct. But, by removing the check, the user would get both a new session and a new token, which would then be in sync.

> When we see a user that has a `jwt_api_auth_token` then maybe it should get converted into the new format with the same expiration and set as `api_auth_token`. That does seem like it could be tricky but it would keep everyone signed in.

Yes, that could work. But I am not sure if we can detect when the session expires at the moment.

> Another alternative would be to entirely log the user out if they don't have an `api_auth_token`.

We though about that, but thought it might be a little brutal.

**Comment by diox** [comment]
Django session backends have `get_expiry_age()` and `get_expiry_date()` but they are actually useless for us because we don't manipulate the session expiry, so it's always set `settings.SESSION_COOKIE_AGE` into the future, which is **not** what we want.

Fortunately, because we're using signed cookies for the session backend, we actually can extract the timestamp and figure out the max age:
```python
session_timestamp = request.session.session_key.split(':')[1]
cookie_max_age = int(baseconv.base62.decode(session_timestamp) + 
    settings.SESSION_COOKIE_AGE - time.time())
```

It's not great, but I think it's good enough for this particular problem.

**Comment by diox** [comment]
> Fortunately, because we're using signed cookies for the session backend, we actually can extract the timestamp and figure out the max age

I've played with that locally but I had to revert it, because although we can set up the expiry on the new cookie to be the right one, out of the box we can't do that with django signing classes - it just uses the current time for the timestamp.

I thought about this more and decided to simply refresh the API token - the user does go through FxA, in a way they just re-authenticated, so I don't think it's a problem to give them a fresh token and cookie. The API token might end up expiring later than the session, giving us the opposite problem, but that's fine, since addons-server ignores that token entirely - worst thing that can happen is that in a little less than 30 days affected users will be still logged in on frontend but not when they go to say, devhub. We can live with that.
